### PR TITLE
Ts-19  58 refactor groupList

### PR DESCRIPTION
**Pull request by Niknikson**

### CODE REVIEW

File: frontend/src/actions/groups.js [hunk 1]
```diff
@@ -1,74 +1,96 @@
...
+export const showAllGroupsSuccess = (res) => {
     return {
-        type: actionTypes.SHOW_ALL_GROUPS,
+        type: actionTypes.SHOW_ALL_GROUPS_SUCCESS,
         result: res,
     };
 };
 
-export const deleteGroupSusses = (res) => {
+export const fetchGroupByIdSuccess = (group) => {
```
Comment (ViktoriiaKosovan) [inline frontend/src/actions/groups.js]: maybe 'fetch' change to get

File: frontend/src/containers/GroupPage/GroupForm.js [hunk 2]
```diff
@@ -2,12 +2,13 @@ import { connect } from 'react-redux';
 import { reduxForm } from 'redux-form';
 import { GROUP_FORM } from '../../constants/reduxForms';
 import { AddGroup } from '../../components/AddGroupForm/AddGroupForm';
+import { submitGroupStart, clearGroupStart } from '../../actions/groups';
 
 const mapStateToProps = (state) => ({
-    group: state.groups.group,
+    //group: state.groups.group,
 });
```
Comment (ViktoriiaKosovan) [inline frontend/src/containers/GroupPage/GroupForm.js]: delete it

File: frontend/src/components/Students/StudentsTable.js [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+        checkAllStudentsSuccess(currentStudentsOnList, checkedAll);
+    };
+
+    const isCheckedAll = () => {
+        setCheckedAll(currentStudentsOnList.every((item) => item.checked));
+    };
+
+    const checkStudent = (event) => {
+        const { value, checked } = event.target;
+        const checkedStudent = students.filter((student) => student.id === Number(value));
```
Comment (ViktoriiaKosovan) [inline frontend/src/components/Students/StudentsTable.js]: you can use + instead of Number

File: frontend/src/components/GroupPage/GroupList.js [hunk 4]
```diff
@@ -65,13 +63,23 @@ const GroupList = (props) => {
...
+
+    if (loading) {
+        return (
+            <section className="centered-container">
+                <CircularProgress />
+            </section>
+        );
+    }
+
+    if (!visibleGroups.length) {
```
Comment (ViktoriiaKosovan) [inline frontend/src/components/GroupPage/GroupList.js]: Could you use isEmpty from lodash?

File: frontend/src/components/GroupPage/GroupPage.scss [hunk 5]
```diff
@@ -1,62 +1,76 @@
...
   flex-direction: column;
   flex-wrap: wrap;
   align-items: center;
- .students{
-   color: #3f51b5 !important;
-   margin: -15px -8px;
-   font-size: 20px;
- }
+  .students {
+    color: #3f51b5 !important;
```
Comment (ViktoriiaKosovan) [inline frontend/src/components/GroupPage/GroupPage.scss]: Could you not to use 'important' in this style?

File: frontend/src/components/Students/StudentTable.scss [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+}
+.edit-cell{
+  display: flex;
+  flex-direction: row;
+  flex-wrap: nowrap;
+  align-items: center;
+  align-content: center;
+  margin: 0 20px;
+  .edit-button{
+    //color: #ade6c6;
```
Comment (ViktoriiaKosovan) [inline frontend/src/components/Students/StudentTable.scss]: delete it

File: frontend/src/containers/TemporarySchedule/TemporarySchedule.js [hunk 7]
```diff
@@ -36,7 +36,7 @@ const TemporarySchedule = (props) => {
         showAllSubjectsService();
         getClassScheduleListService(null);
         getLessonTypes();
-        showAllGroupsService();
+        //fetchEnabledGroupsStart();  rename to fetchEnabledGroupsStart
```
Comment (ViktoriiaKosovan) [inline frontend/src/containers/TemporarySchedule/TemporarySchedule.js]: Do you need this comment ?

File: frontend/src/actions/groups.js (hunk 1 above)
Comment (Niknikson) [reply frontend/src/actions/groups.js]: fixed

File: frontend/src/containers/TemporarySchedule/TemporarySchedule.js (hunk 7 above)
Comment (Niknikson) [reply frontend/src/containers/TemporarySchedule/TemporarySchedule.js]: fixed

File: frontend/src/components/Students/StudentTable.scss (hunk 6 above)
Comment (Niknikson) [reply frontend/src/components/Students/StudentTable.scss]: done

File: frontend/src/components/GroupPage/GroupList.js [hunk 8]
```diff
@@ -64,14 +67,39 @@ const GroupList = (props) => {
...
     };
-    const onDeleteStudent = (id) => {
-        deleteStudentStart(id);
+
+    const checkParamsForActions = () => {
+        const { id, action } = match.params;
+
+        const checkParamsAndSetActions = {
+            'add-student': showAddStudentDialog,
+            'show-students': showStudentsByGroup,
```
Comment (Niknikson) [inline frontend/src/components/GroupPage/GroupList.js:86]: need to create constants ?

File: frontend/src/containers/DepartmentPage/DepartmentPage.js [hunk 9]
```diff
@@ -73,7 +73,7 @@ const DepartmentPage = (props) => {
         : search(enabledDepartments, term, ['name']);
 
     const submitAddForm = (data) => {
-        return data.id ? createDepartmentService(data) : updateDepartmentService(data);
+        return data.id ? updateDepartmentService(data) : createDepartmentService(data);
```
Comment (Niknikson) [inline frontend/src/containers/DepartmentPage/DepartmentPage.js:76]: this is not mine

File: frontend/src/services/groupService.js [hunk 10]
```diff
@@ -3,7 +3,7 @@ import { store } from '../store';
 import axios from '../helper/axios';
```
Comment (HiroFive) [inline frontend/src/services/groupService.js:3]: Do we still need this service?

File: frontend/src/services/uploadFile.js [hunk 11]
```diff
@@ -12,19 +12,21 @@ import {
...
                 students !== 1
                     ? ` ${i18n.t(FORM_STUDENTS_FILE_LABEL)}`
                     : ` ${i18n.t(FORM_STUDENT_FILE_LABEL)}`;
             successHandler(
                 i18n.t(FILE_BACK_END_SUCCESS_OPERATION, {
-                    cardType: students,
+                    cardType: i18n.t(FORM_STUDENTS_FILE_LABEL),
                     actionType: i18n.t(FILE_LABEL),
                 }),
             );
```
Comment (HiroFive) [inline frontend/src/services/uploadFile.js]: maybe you should create saga for this request

File: frontend/src/share/SearchPanel/SearchPanel.scss [hunk 12]
```diff
@@ -1,15 +1,23 @@
...
+@import '../../styles/mixin.scss';
 
-.search-group {
-    min-height: 85px;
+// .search-group {
+//     @include displayCenter();
+//     .MuiTypography-body1{
+//         font-size: 0.8rem;
+//     }
+// }
```
Comment (HiroFive) [inline frontend/src/share/SearchPanel/SearchPanel.scss]: if you don't need this line, delete it

Comment (Niknikson) [reply frontend/src/share/SearchPanel/SearchPanel.scss]: done

File: frontend/src/services/uploadFile.js (hunk 11 above)
Comment (Niknikson) [reply frontend/src/services/uploadFile.js]: done

File: frontend/src/services/groupService.js (hunk 10 above)
Comment (Niknikson) [reply frontend/src/services/groupService.js:3]: done

**Review by HiroFive — APPROVED** [review]

**Review by Yashchemskyi-Oleksandr — APPROVED** [review]

**Pull request closed without merging**